students/hyska_monika/lesson_11_argparse_math/regex_search.py

```diff
@@ -0,0 +1,60 @@
+"""
+Program that opens all .txt files in a folder and
+searches for any line that matches a user-supplied regular expression.
+The results should be printed to the screen.
+"""
+
+import re
+import os
+import argparse
+import sys
+
+
+def check_arg(args=None):
+    parser = argparse.ArgumentParser(description='Search regex on files')
+    parser.add_argument('-p', '--path',
+                        help='path to search',
+                        default='./')
+    parser.add_argument('-r', '--regex',
+                        required='True',
+                        help='regex to search')
+    results = parser.parse_args(args)
+    return (results.path,
+            results.regex)
+
+
+def files_in_path(path):
+    only_files = [f for f in os.listdir(path)
+                  if os.path.isfile(os.path.join(path, f))]
+    return only_files
+
+
+def search_regex(path, regex):
+    files = files_in_path(path)
+    pattern = re.compile(regex)
+    print("Lines contain searched regex:")
```

Comment:
English: `Lines containing regex`, this just doesn't sound right to me, but I can't explain it well: https://english.stackexchange.com/a/133354